src/controllers/auth.ts

```diff
@@ -0,0 +1,235 @@
+import { NextFunction, Request, Response } from "express";
+import userModel, { IUser } from "../models/user";
+import bcrypt from "bcrypt";
+import jwt from "jsonwebtoken";
+import { Document } from "mongoose";
+import { refreshTokenExpires, tokenExpires, tokenSecret } from "../config";
+
+// eslint-disable-next-line max-statements
+const register = async (req: Request, res: Response) => {
+    try {
+        const existingUserByEmail = await userModel.findOne({ email: req.body.email });
+        if (existingUserByEmail) {
+            const CONFLICT_STATUS = 409;
+            res.status(CONFLICT_STATUS).send("Email already in use");
+            return;
+        }
+
+        const existingUserByUsername = await userModel.findOne({ username: req.body.username });
+        if (existingUserByUsername) {
+            const CONFLICT_STATUS = 409;
+            res.status(CONFLICT_STATUS).send("Username already in use");
+            return;
+        }
```

Comment:
It is not necessary to check duplicates since an index is defined with unique values for username and email, and mongoose will throw an appropriate error in such case.
The code that is responsible for handling this error is in base-controller.ts (`DBHandler`).